test/persistence-hooks.suite.js

```diff
@@ -2946,6 +2946,431 @@ module.exports = function(dataSource, should, connectorCapabilities) {
 
     });
 
+    describe('PersistedModel.upsertWithWhere', function() {
+
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ extra: 'not-found' },
+          { id: 'not-found', name: 'not found', extra: 'not-found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            if (dataSource.connector.upsertWithWhere) {
```

Comment:
Now that we get the same order of operation hooks for `atomic` Vs `non-atomic` implementation, we should have just one branch and `if` is not necessary. This applies for other places as well.